### ISSUE TITLE
[Feat] automatic computation of class weights for imbalanced classification

### ISSUE THREAD

**Issue by pietrolesci** [PyTorchLightning/lightning-flash#873]
## 🚀 Feature
Add support for automatic class weights calculation

## Motivation
Important features in imbalanced classification tasks (especially in the few-shot regime)

### On the user side

```python
dm = TextClassificationData.from_csv(
    train_file="data/imdb/train.csv",
    val_file="data/imdb/valid.csv",
    test_file="data/imdb/test.csv",
    input_fields="review",
    target_fields="sentiment",
    backbone="prajjwal1/bert-tiny",
    class_weights: Optional[Union[Dict[str, float], str]] = None
)

model = TextClassifier(
    bakcbone=dm.backbone,
    num_classes=dm.num_classes,
    class_weights=dm.class_weights,
)
```
With functionalities like in [scikit-learn implementation](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression)
```
class_weight: dict or ‘balanced’, default=None
- Weights associated with classes in the form {class_label: weight}. If not given, all classes are supposed to have weight one.
- The “balanced” mode uses the values of y to automatically adjust weights inversely proportional to class frequencies in the input data as n_samples / (n_classes * np.bincount(y)).
```

### In the backend
- Add functionality to compute the weights
- Pass the weights to the underlying cross-entropy loss

## Affected parts of the code
- `ClassificationMixin` for the model side
- Individual data sources (e.g., `TextDataSource`
